From: Vojtech Trefny <vtrefny@redhat.com>
Date: Wed, 26 Jul 2023 10:12:49 +0200
Subject: fs: Fix memory leak

We can use g_dir_make_tmp instead of g_build_path and g_mkdtemp
and avoid a leaking string in the process.
---
 src/plugins/fs/generic.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/plugins/fs/generic.c b/src/plugins/fs/generic.c
index 0703eda..77e5496 100644
--- a/src/plugins/fs/generic.c
+++ b/src/plugins/fs/generic.c
@@ -639,8 +639,7 @@ static gchar* fs_mount (const gchar *device, gchar *fstype, gboolean read_only,
     if (!mountpoint) {
         if (l_error == NULL) {
             /* device is not mounted -- we need to mount it */
-            mountpoint = g_build_path (G_DIR_SEPARATOR_S, g_get_tmp_dir (), "blockdev.XXXXXX", NULL);
-            mountpoint = g_mkdtemp (mountpoint);
+            mountpoint = g_dir_make_tmp ("blockdev.XXXXXX", NULL);
             if (!mountpoint) {
                 g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_FAIL,
                              "Failed to create temporary directory for mounting '%s'.", device);
